src/utils/Subscription.js

```diff
@@ -1,45 +1,61 @@
 // encapsulates the subscription logic for connecting a component to the redux store, as
 // well as nesting subscriptions of descendant components, so that we can ensure the
 // ancestor components re-render before descendants
+
+function initListeners() {
+  let count = 0
+  let current = []
+  let next = []
+
+  return {
+    clear() {
+      count = 0
+      next = null
+      current = null
+    },
+
+    notify() {
+      current = next
+      for (let i = 0; i < count; i++) {
```

Comment:
Please remove this `count` variable. It risks things getting out of sync somehow. Calling `.length` is just as fast, especially in engines like v8 that use hidden classes.